packages/jest-cli/src/watch.js

```diff
@@ -126,10 +133,20 @@ const watch = (
       results => {
         isRunning = false;
         hasSnapshotFailure = !!results.snapshot.failure;
+        failedSnapshotTestPaths = getFailedSnapshotTests(results);
```

Comment:
This function filters the test paths based on failed snapshot flag, that's why this tool now works on _per file_ basis.
We could get each test using similar logic as coded in `testNamePatternPrompt`. You should be able to reuse it and feed this to `snapshotInteractiveMode`.